tests/unit/manager/default/test_background_job.py

```diff
@@ -677,7 +677,7 @@ def test_delete_http_policy_cert_not_deployed(self):
 
     def test_delete_http_policy_cert_doesnt_exist(self):
         self.mock_storage.certificates_controller.\
-            get_certs_by_domain.return_value = []
+            get_certs_by_domain.side_effect = ValueError
```

Comment:
For this test (and other tests like it), it would be better to use assert raises decorator: `@raises(ValueError)`.

Reply:
Could do that if the test is written only to call one method and is expected to receive an exception. However, several methods are called (or mocked for return values/exceptions). Will try and change wherever this is possible(assert exceptions)